meta-resin-common/classes/binary-compress.bbclass

```diff
@@ -0,0 +1,28 @@
+# e.g.  COMPRESS_BINARY = "path/bin1 path/bin2 path/bin3"
+# Define what binaries that we find in the package tree will be compress
+# This variable's definition is MANDATORY if this class is used
+# NOTE: path is relative to PKGD directory
+COMPRESS_BINARY ?= ""
+
+DEPENDS_append = " bash upx"
+
+find_and_compress() {
+    #Sanity check
+    if [ -z ${COMPRESS_BINARY} ]; then
+        bbwarn "Binary compress class imported but COMPRESS_BINARY variable was found empty. Skipping compression."
+    else
+        #Find all executables once
+        pkg_dir=${PKGD}
+        execs=$(find $pkg_dir -executable -type f)
+        #Compress
+        for exec in $execs; do
+            bin=${exec#$pkg_dir}
```

Comment:
The behavior of the #, ##, %, and %% operators are unspecified by POSIX and the ksh88 manual when used together with the @ or * parameters. Dash applies the trimming to the flattened result. mksh/pdksh treats it as a bad expansion.

http://mywiki.wooledge.org/Bashism